pkg/build/metrics/prometheus/metrics.go

```diff
@@ -77,40 +96,58 @@ func (bc *buildCollector) Collect(ch chan<- prometheus.Metric) {
 
 	// since we do not collect terminal build metrics on a per build basis, collectBuild will return counts
 	// to be added to the total amount posted to prometheus
-	var failed, error, cancelled, complete int
+	var error, cancelled, complete int
+	reasons := map[string]int{}
 	for _, b := range result {
-		f, e, cc, cp := bc.collectBuild(ch, b)
-		failed = failed + f
+		e, cc, cp, r := bc.collectBuild(ch, b)
+		for key, value := range r {
+			count := reasons[key]
+			count = count + value
+			reasons[key] = count
+		}
 		error = error + e
 		cancelled = cancelled + cc
 		complete = complete + cp
 	}
-	addCountGauge(ch, terminalBuildCountDesc, failedPhase, float64(failed))
+	// explicitly note there are no failed builds
+	if len(reasons) == 0 {
+		addCountGauge(ch, failedBuildCountDesc, "", float64(0))
+	}
+	for reason, count := range reasons {
+		addCountGauge(ch, failedBuildCountDesc, reason, float64(count))
+	}
 	addCountGauge(ch, terminalBuildCountDesc, errorPhase, float64(error))
```

Comment:
these should probably be counted under failedBuilds with a reason of "Build pod encountered an error" or something.

Reply:
(errored builds are closer to "failed" builds than they are to "completed+canceled" builds).

Reply:
yeah I went back and forth on this, considering each point you just mentioned vs. the fact that in the runtime, following a guru search, the StatusReason field in the build is never set with an `error` build

bottom line, I'm not convinced ... unless there is more than one reason for builds in `error` state, it is only wasteful to add a label reason for it ... we shouldn't use prometheus labels to clarify like this

Reply:
I'm saying we treat "Error" as the reason.  Treat Failed and Error builds the same, when applying the reason label for the Error builds, just manufacture a reason.

In other words i want the "failedBuilds" metrics to include the count of Error builds, because otherwise they may be ignored under "terminated" builds as not interesting.

Reply:
I see ... ok, we should make the reason string consistent in look/feel to the others ... I'll go with `BuildPodError` .... there already `BuildPodDeleted` and `BuildPodExists`

Reply:
sounds good.